src/ex4/server/services/item_manager.js

```diff
@@ -1,89 +1,60 @@
+const { Op } = require('sequelize');
+const { Item } = require('../../db/models');
+const { v4: uuidv4 } = require('uuid');
 const pokemonClient = new pokemonApi();
-const cache = new Map();
 
-async function addTasksOrPokemons(tasks) {
-  let data = await readTasksFile();
-  if (!data) {
-    data = [];
+class ItemManager {
+  constructor() {
+    this.pokemonClient = new pokemonApi();
   }
 
-  const cachedTasks = tasks.filter((task) => !isNaN(task) && cache.has(task)); // find cached tasks (only numbers)
-
-  const result = await pokemonClient.fetchPokemons(
-    tasks.filter((item) => !cache.has(item))
-  ); // fetch only new requests
-
-  const modifiedResult = result.map((item) => {
-    if (typeof item === 'object') {
-      return { text: item.task };
-    }
-    return { text: item };
-  });
-
-  const modifiedCache = cachedTasks.map((task) => {
-    return { text: cache.get(task) };
-  });
-
-  data.push(...modifiedResult, ...modifiedCache);
-  await writeToTasksFile(data);
-
-  updateCache(
-    tasks,
-    result.filter((res) => typeof res === 'object')
-  ); //add new pokemon requests to the cache
-}
-
-async function deleteTask(task) {
-  const fileData = await getAll(); // reading file data
-  const taskIndex = fileData.findIndex((value) => value.text === task.text);
-  if (taskIndex === -1) {
-    return;
-  }
-  fileData.splice(taskIndex, 1);
-  await writeToTasksFile(fileData);
-  return task;
-}
-
-async function deleteAll() {
-  await writeToTasksFile([]);
-}
-
-async function getAll() {
-  return await readTasksFile();
-}
-
-async function readTasksFile() {
-  try {
-    const data = await fs.readFile(tasksFile);
-    return JSON.parse(data.toString());
-  } catch (error) {
-    console.error(`Got an error trying to read the file: ${error.message}`);
-  }
-}
-
-async function writeToTasksFile(content) {
-  try {
-    await fs.writeFile(tasksFile, JSON.stringify(content));
-  } catch (error) {
-    console.error(`Failed to write to file ${error.message}`);
-  }
-}
-
-function updateCache(keys, values) {
-  keys.forEach((key) => {
-    if (!isNaN(key) && !cache.has(key)) {
-      const value = values.find((val) => parseInt(key) === val.id);
-      cache.set(key, value.task);
-    }
-  });
+  getAll = async () => await Item.findAll({ order: [['createdAt', 'DESC']] });
+
+  deleteAll = async () =>
+    await Item.destroy({
+      truncate: true,
+    });
+
+  deleteItem = async (id) =>
+    await Item.destroy({
+      where: {
+        Id: id,
+      },
+    });
+
+  changeState = async (id) => {
+    const item = await Item.findOne({ where: { Id: id } });
+    const newStatus = !item.status;
+    await Item.update({ status: newStatus }, { where: { Id: id } });
+  };
+
+  editItem = async (id, text) => {
+    await Item.update({ pokemonId: null, ...text }, { where: { Id: id } });
+  };
+  addTasksOrPokemons = async (tasks) => {
+    const pokemonsRegisterd = await Item.findAll({
+      where: { pokemonId: { [Op.ne]: null } },
```

Comment:
I'm not fetching pokemons if I find them on the table by pokemonId.
But if the user will edit a "Catch pokemon" task to something else (like "hello"), it might not be pokemon anymore.
I remove the pokemonId so this Id won't be associated with the wrong task...